jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IO.java

```diff
@@ -224,36 +224,110 @@ public static void copyDir(File from, File to) throws IOException
     }
 
     /**
-     * Copy the contents of a directory from one directory to another.
+     * Copy the contents of a source directory to destination directory.
      *
      * @param srcDir the source directory
      * @param destDir the destination directory
      * @throws IOException if unable to copy the file
      */
+    public static void copyDir(Path srcDir, Path destDir) throws IOException
+    {
+        if (!Files.isDirectory(Objects.requireNonNull(srcDir)))
+            throw new IllegalArgumentException("Source is not a directory: " + srcDir);
+        if (!Files.isDirectory(Objects.requireNonNull(destDir)))
+            throw new IllegalArgumentException("Dest is not a directory: " + destDir);
+
+        try (Stream<Path> sourceStream = Files.walk(srcDir))
+        {
+            Iterator<Path> iterFiles = sourceStream
+                .filter(Files::isRegularFile)
+                .iterator();
+            while (iterFiles.hasNext())
+            {
+                Path sourceFile = iterFiles.next();
+                Path relative = srcDir.relativize(sourceFile);
+                Path destFile = resolvePath(destDir, relative);
+                if (!Files.exists(destFile.getParent()))
+                    Files.createDirectories(destFile.getParent());
+                copyFile(sourceFile, destFile);
+            }
+        }
+    }
+
+    /**
+     * Copy the contents of a source directory to destination directory.
+     *
+     * @param srcDir the source directory
+     * @param destDir the destination directory
+     * @param copyOptions the options to use on the {@link Files#copy(Path, Path, CopyOption...)} commands.
+     * @throws IOException if unable to copy the file
+     */
     public static void copyDir(Path srcDir, Path destDir, CopyOption... copyOptions) throws IOException
     {
         if (!Files.isDirectory(Objects.requireNonNull(srcDir)))
             throw new IllegalArgumentException("Source is not a directory: " + srcDir);
         if (!Files.isDirectory(Objects.requireNonNull(destDir)))
             throw new IllegalArgumentException("Dest is not a directory: " + destDir);
 
-        try (Stream<Path> sourceStream = Files.walk(srcDir, 20))
+        try (Stream<Path> sourceStream = Files.walk(srcDir))
```

Comment:
Are you certain removing this limit won't have any adverse effects?

Reply:
This was an arbitrary limit with no real value.
With the changes in this PR, the copy of a war contents to work directory will stress the original limit.
Having this limit also introduced a silent bug. Deeper directories were just not copied without a log statement.